src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsAbcdProductsPriority.java

```diff
@@ -0,0 +1,57 @@
+package com.wikia.webdriver.testcases.adstests;
+
+import com.wikia.webdriver.common.contentpatterns.AdsContent;
+import com.wikia.webdriver.common.core.Assertion;
+import com.wikia.webdriver.common.templates.TemplateNoFirstLoad;
+import com.wikia.webdriver.pageobjectsfactory.pageobject.adsbase.AdsBaseObject;
+import org.testng.annotations.Test;
+
+public class TestAdsAbcdProductsPriority extends TemplateNoFirstLoad {
+
+  private static final String WIKIA = "project43";
+  private static final String URL_ARTICLE = "/SyntheticTests/UAP/ABCD/ProductsPriority";
+  private static final String URL_ARTICLE_WITH_DEBUG_MODE = URL_ARTICLE + "?appnexusast_debug_mode";
+
+
+  @Test(
+      groups = "AbcdProductPriorityOasis"
+  )
+  public void adsAppNexusVideoAdIsDisplayedIfAbcdAdIsTargetedOnThisSameArticleOasis() {
+    AdsBaseObject ads = new AdsBaseObject(driver, urlBuilder.getUrlForPage(WIKIA, URL_ARTICLE_WITH_DEBUG_MODE));
+    verifyAppNexussAdIsDisplayed(ads);
+    verifyAbcdAdIsNotDisplayed(ads);
+  }
+
+  @Test(
+      groups = "AbcdProductPriorityOasis"
+  )
+  public void adsAbcdAdIsDisplayedIfBidersAreDisabledOasis() {
+    AdsBaseObject ads = new AdsBaseObject(driver, urlBuilder.getUrlForPage(WIKIA, URL_ARTICLE));
+    verifyAbcdAdIsDisplayed(ads);
+  }
+
+  private void verifyAppNexussAdIsDisplayed(AdsBaseObject ads){
+    ads.scrollToSlot(AdsContent.INCONTENT_PLAYER);
+
+    Assertion.assertTrue(ads.getGptParams(AdsContent.INCONTENT_PLAYER, "data-gpt-slot-params").contains("appnexusAst"),
+        "AppNexus ad param: appnexusAst is not present");
+    Assertion.assertTrue(ads.getGptParams(AdsContent.INCONTENT_PLAYER, "data-gpt-line-item-id").equals("4406759476"),
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Move the "4406759476" string literal on the left side of this string comparison. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AS1132)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Define a constant instead of duplicating this literal "data-gpt-line-item-id" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AS1192)